src/main/java/com/epam/ta/reportportal/core/launch/impl/GetLaunchHandler.java

```diff
@@ -56,23 +59,21 @@
 @Service
 public class GetLaunchHandler /*extends StatisticBasedContentLoader*/ implements IGetLaunchHandler {
 
-	//	private ProjectRepository projectRepository;
+	private final LaunchRepository launchRepository;
+	private final LaunchTagRepository launchTagRepository;
 
-	@Autowired
-	private LaunchRepository launchRepository;
+	public GetLaunchHandler(LaunchRepository launchRepository, LaunchTagRepository launchTagRepository) {
 
-	//	@Autowired
-	//	public void setProjectRepository(ProjectRepository projectRepository) {
-	//		this.projectRepository = projectRepository;
-	//	}
+		this.launchRepository = launchRepository;
+		this.launchTagRepository = launchTagRepository;
+	}
 
 	@Override
 	public LaunchResource getLaunch(Long launchId, ReportPortalUser.ProjectDetails projectDetails, ReportPortalUser user) {
 		Launch launch = launchRepository.findById(launchId)
 				.orElseThrow(() -> new ReportPortalException(ErrorType.LAUNCH_NOT_FOUND, launchId));
 
-		expect(launch.getProjectId(), equalTo(projectDetails.getProjectId())).verify(
-				ErrorType.FORBIDDEN_OPERATION,
+		expect(launch.getProjectId(), equalTo(projectDetails.getProjectId())).verify(ErrorType.FORBIDDEN_OPERATION,
 				formattedSupplier("Specified launch with id '{}' not referenced to specified project '{}'", launchId, projectDetails)
 		);
 
```

Comment:
Should we check that?

Reply:
does it work?) I suppose somewhere in controller it should be like:
@AuthenticationPrincipal ReportPortalUser user
ProjectUtils.extractProjectDetails(user, projectName)
which means you don't need to validate it since if there is no such project, exception will be thrown

Reply:
agree with andrei

Reply:
it is checked in controller while extracting project details